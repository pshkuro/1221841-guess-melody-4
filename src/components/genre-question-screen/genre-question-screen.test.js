import React from "react";
import renderer from "react-test-renderer";
import GenreQuestionScreen from "./genre-question-screen.jsx";

const question = {
  type: `genre`,
  genre: `rock`,
  answers: [{
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
    id: 11,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `blues`,
    id: 33,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `jazz`,
    id: 89,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
    id: 12,
  }]
};


it(`Render GenreQuestionScreen`, () => {
  const onAnswerMock = jest.fn();
  const onRenderPlayerMock = jest.fn();

  const tree = renderer
  .create(
      <GenreQuestionScreen
        onAnswer={onAnswerMock}
        question={question}
        renderPlayer={onRenderPlayerMock}
      />,
      {createNodeMock: () => {
        return {};
      }
      });

  expect(tree).toMatchSnapshot();
});
