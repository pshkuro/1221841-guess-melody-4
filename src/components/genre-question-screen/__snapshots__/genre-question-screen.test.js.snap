// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenreQuestionScreen 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-11"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-11"
        />
        <label
          className="game__check"
          htmlFor="answer-11"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-33"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-33"
        />
        <label
          className="game__check"
          htmlFor="answer-33"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-89"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-89"
        />
        <label
          className="game__check"
          htmlFor="answer-89"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="answer-12"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-12"
        />
        <label
          className="game__check"
          htmlFor="answer-12"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
